### PR TITLE
NTV-612: Move Minimum Version Up To iOS 14

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Updates minimum iOS deployment target to 14.0.

# 🤔 Why

With the release of iOS 16, let's keep our usual support of two minimum versions back on iOS.

# 🛠 How

1.  Set the minimum deployment target in the project settings, address deprecation warnings.
2. Nothing new that we implement was deprecated in iOS 16 specifically. Just a few things in iOS 14 (see code comments)

# ✅ Acceptance criteria

- [x] App runs on iOS 14+
- [x] No regressions 

# **Notes**
In the [last calendar year](https://console.firebase.google.com/u/0/project/kickstarter-ios/analytics/app/ios:com.kickstarter.kickstarter/audience/~2Foverview%3Ft%3D1666711125188&fpn%3D553793001056&swu%3D1&sgu%3D1&sus%3Dupgraded&cs%3Dapp.m.audiences.overview&g%3D1?hl=en) we've had:
- Engaged sessions, "The number of sessions that lasted longer than 10 seconds, or had a conversion event, or had 2 or more screen or page views",  by iOS 14.8+ users
- Total number of active users at iOS version 14.6+

So, we shouldn't see any users being dropped from this update.

see our [firebase console](https://console.firebase.google.com/u/0/project/kickstarter-ios/analytics/app/ios:com.kickstarter.kickstarter/audience/~2Foverview%3Ft%3D1666711125188&fpn%3D553793001056&swu%3D1&sgu%3D1&sus%3Dupgraded&cs%3Dapp.m.audiences.overview&g%3D1?hl=en) for more current data points.

### CODE REVIEW

File: KsApi/models/graphql/adapters/User+UserFragment.swift
Comment (scottkicks) [inline KsApi/models/graphql/adapters/User+UserFragment.swift:82]: Removing unused property

File: Kickstarter-iOS/AppDelegate.swift
Comment (scottkicks) [inline Kickstarter-iOS/AppDelegate.swift:499]: `.alert` was deprecated in iOS 14. The docs say to now use `.banner` and `.list` https://developer.apple.com/documentation/usernotifications/unnotificationpresentationoptions/1649506-alert

`.list` shows notifications in Notification Center.
`.banner` presents notifications as a banner.

File: Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift
Comment (scottkicks) [inline Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift:225]: [`requestReview(in: UIWindowScene)`](https://developer.apple.com/documentation/storekit/skstorereviewcontroller/2851536-requestreview) should now be used instead of `requestReview()`

Comment (msadoon) [reply Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift:225]: Yea cool, just when making these types of StackOverflow suggested changes be careful to test that the review dialog does show up, because `UIApplication.shared.connectedScenes` is a bit ambiguous. I tested though and it shows.

**Review by msadoon — CHANGES_REQUESTED** [review]
Overall looks good, just make a few changes as suggested.

File: Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift
Comment (scottkicks) [reply Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift:225]: Absolutely. I was unsure of this at first as well, and made sure to find other suggestions, in apple forums for example, before deciding to go with this.

**Comment by msadoon** [comment]
just wait until tests pass then merge

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Features/ThanksProjects/Controller/ThanksViewController.swift:223:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**
